tutor/src/models/task-plans/teacher/plan.js

```diff
@@ -364,6 +364,10 @@ class TeacherTaskPlan extends BaseModel {
     return get(this, 'settings.page_ids', []);
   }
 
+  @computed get assignedSections() {
+    return this.course.referenceBook.pages.all.filter(page => this.pageIds.includes(page.id));
+  }
```

Comment:
How about `return compact(his.pageIds.map(pgId => this.course.referenceBook.pages.byId.get(pgId))`

That will be sorted correctly since sometimes the teachers do re-order sections on an assignment.

I also think that'll be a bit more efficient since the pages.byId is pre-computed and that won't need to look at every single page id.  Some books are fairly large and have a lot of sections.

I'm not sure if the lodash `compact()` is really needed but there might be some cases where a page isn't found.

Reply:
this is good. it is faster and it is not going through all the pages in a book. 

I think we might need `compact` just in case a weird falsey value gets returned.